php/Terminus/Models/Environment.php

```diff
@@ -407,6 +407,63 @@ public function getConnectionMode() {
   }
 
   /**
+   * Retrieves recommended DNS settings for this environment
+   *
+   * @return array[]
+   * @throws TerminusException
+   */
+  public function getDns() {
+    if ($this->site->get('service_level') == 'free') {
+      throw new TerminusException(
+        'To add custom domains, this site must be upgraded to a paid plan.',
+        [],
+        1
+      );
+    }
+    $response = $this->request->simpleRequest(
+      sprintf(
+        'sites/%s/environments/%s?environment_state=true',
+        $this->site->get('id'),
+        $this->get('id')
+      )
+    );
+    $hosts          = $response['data']->urls;
+    $base_names     = $this->determineBaseNames($hosts);
+    $load_balancers = (array)$response['data']->loadbalancers;
+    $load_balancer  = array_shift($load_balancers);
+    $dns_settings   = [];
+    foreach ($hosts as $host) {
+      preg_match("/pantheon.io$/", $host, $matches);
+      if (empty($matches)) {
+        if (in_array($host, $base_names)) {
+          $dns_settings[] = [
+            'host'        => $host,
+            'record_type' => 'A',
+            'value'       => $load_balancer->ipv4,
+          ];
+          $dns_settings[] = [
+            'host'        => $host,
+            'record_type' => 'AAAA',
+            'value'       => $load_balancer->ipv6,
+          ];
+        } else {
+          $dns_settings[] = [
+            'host'        => $host,
+            'record_type' => 'CNAME',
+            'value'       => sprintf(
+              '%s-%s.pantheon.io',
+              $this->get('id'),
+              $this->site->get('name')
+            )
+          ];
+        }
+      }
+    }
```

Comment:
@kibra pointed out that my spec above doesn't define what "base name" means. Here's the js code for that:

```
baseDomain = 'gotpantheon.com'
if environment.get('dns_zone')?
  baseDomain = environment.get('dns_zone')
if site.get('base_domain')?
  baseDomain = site.get('base_domain')
baseDomain
```